### PR TITLE
Remove ESP-IDF submodule

### PR DESCRIPTION
**Pull request by renzenicolai**

### CODE REVIEW

**Comment by Pwuts** [comment]
Nice :)

**Comment by smunaut** [comment]
Why ? I don't see how it makes much of a difference.

File: Makefile [hunk 1]
```diff
@@ -6,11 +6,13 @@ SHELL := /usr/bin/env bash
...
 .PHONY: prepare clean build flash erase monitor menuconfig image qemu install size size-components size-files format
 
-all: prepare build flash
+all: build flash
 
 prepare:
 	git submodule update --init --recursive
-	cd esp-idf; bash install.sh
+	rm -rf "$(IDF_PATH)"
+	git clone --recursive --branch v4.4.1 https://github.com/espressif/esp-idf.git
```
Comment (Pwuts) [inline Makefile:14]: Should this clone explicitly into `${IDF_PATH}`?

**Comment by Pwuts** [comment]
On top of not including it as a submodule, we could make the install of the ESP-IDF optional, how about that? If a developer already has a compatible IDF version, it makes sense to just let them use that one, no? We could either check the environment for a present IDF and check its version, or prompt the user whether to install the IDF in the workspace.

**Pull request merged**